### PR TITLE
Redirect user to detail form if no legal address

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
N/A

#### What's this PR do?
Prevents OFAC errors when logging in via edx under various conditions

#### How should this be manually tested?
Copy over the CYBERSOURCE_ and MAILGUN_ env values from CI to enable compliance checks and verification emails.

Go through the following steps with `FEATURE_SOCIAL_AUTH_API=False`:
- Log in via edx as an existing user who already has a bootcamp account
- Log in via edx as a user who exists in edx but not bootcamp
- Log in via edx as a new user who does not exist on edx or bootcamp

You should be redirected to `./pay` after login for all the above.  

Repeat the above steps with `FEATURE_SOCIAL_AUTH_API=True`:
- You should be redirected to `./create-account/details` for all the above
- Complete the registration forms and then you should be redirected to `./pay`

Log out.

Go to `../create-account` and submit the form, wait for the verification email, click the link, step through the registration forms.  You should be redirected to `../pay` at the end.

### CODE REVIEW

**Comment by rhysyngsun** [comment]
I'm taking a personal day tomorrow, so:

@pdpinch @mbertrand I've functionally tested this and it looks good, but it needs more test coverage

@gsidebo this includes a change to `Profile.is_complete` which some of your application code depends on, i've updated those tests but worth you doing a onceover

**Comment by noisecapella** [comment]
Code looks good to me. I succeeded in creating a new user but I want to try the other scenarios to make sure everything is working.

**Comment by noisecapella** [comment]
LGTM. I'll hand it over to @mbertrand

**Comment by mbertrand** [comment]
@rhysyngsun deferring to you on whether my changes to this PR make sense.

File: authentication/pipeline/user.py [hunk 1]
```diff
@@ -86,7 +86,16 @@ def create_user_via_email(
     Raises:
         RequirePasswordAndPersonalInfoException: if the user hasn't set password or name
     """
-    if user or not strategy.is_api_enabled():
+    if not strategy.is_api_enabled():
+        return {}
+
+    # if the user has data for all the fields this step
```
Comment (annagav) [inline authentication/pipeline/user.py:92]: "skip this step"

**Comment by rhysyngsun** [comment]
Aside from Anna's comment, lgtm.

**Pull request merged**